IFTPipeline.jl/src/preprocess.jl

```diff
@@ -296,5 +296,194 @@ function preprocess_single(; truecolor::T, falsecolor::T, landmask::T, landmask_
+    ice_labels_ratio_upper=0.75,
+
+    # Adaptive histogram equalization parameters
+    adapthisteq_white_threshold=25.5, 
+    adapthisteq_entropy_threshold=4, 
+    adapthisteq_white_fraction_threshold=0.4,
+
+    # Gamma parameters
+    gamma=1,
+    gamma_factor=1,
+    gamma_threshold=220,
+
+    # Unsharp mask parameters
+    unsharp_mask_radius=10, 
+    unsharp_mask_amount=2.0, 
+    unsharp_mask_factor=255.0,
+
+    # Brighten parameters
+    brighten_factor = 0.1,
+
+    # Preliminary ice mask parameters
+    prelim_icemask_radius=10, 
+    prelim_icemask_amount=2, 
+    prelim_icemask_factor=0.5,
+
+    # Main ice mask parameters
+    icemask_band_7_threshold=5,
+    icemask_band_2_threshold=230,
+    icemask_band_1_threshold=240,
+    icemask_band_7_threshold_relaxed=10,
+    icemask_band_1_threshold_relaxed=190,
+    icemask_possible_ice_threshold=75,
+    icemask_n_clusters=3,
+
+    ) where {T<:AbstractString}
+
+    @info "Processing images: $truecolor, $falsecolor, $landmask_dilated"
+    truecolor_img = loadimg(; dir=dirname(truecolor), fname=basename(truecolor))
+    falsecolor_img = loadimg(; dir=dirname(falsecolor), fname=basename(falsecolor))
+
+    # TODO: make symmetric landmask saving/loading functions
+    landmask_dilated=BitMatrix(FileIO.load(landmask_dilated))
+    
+    # Invert the landmasks – in the tiling version of the code, 
+    # the landmask is expected to be the other polarity compared with
+    # the non-tiling version.
+    landmask = (
+        dilated=map(!, landmask_dilated),
+    )
+
+    @info "Remove alpha channel if it exists"
+    rgb_truecolor_img = RGB.(truecolor_img)
+    rgb_falsecolor_img = RGB.(falsecolor_img)
+
+    @info "Get tile coordinates"
+    tiles = IceFloeTracker.get_tiles(
+        rgb_truecolor_img; 
+        rblocks=tile_rblocks, 
+        cblocks=tile_cblocks
+    )
+    @debug tiles
+
+    @info "Set ice labels thresholds"
+    ice_labels_thresholds = (
+        prelim_threshold=ice_labels_prelim_threshold,
+        band_7_threshold=ice_labels_band_7_threshold,
+        band_2_threshold=ice_labels_band_2_threshold,
+        ratio_lower=ice_labels_ratio_lower,
+        ratio_upper=ice_labels_ratio_upper,
+        use_uint8=true,
+    )
+    @debug ice_labels_thresholds
+
+    @info "Set adaptive histogram parameters"
+    adapthisteq_params = (
+        white_threshold=adapthisteq_white_threshold,
+        entropy_threshold=adapthisteq_entropy_threshold,
+        white_fraction_threshold=adapthisteq_white_fraction_threshold,
+    )
+    @debug adapthisteq_params
+
+    @info "Set gamma parameters"
+    adjust_gamma_params = (
+        gamma=gamma,
+        gamma_factor=gamma_factor,
+        gamma_threshold=gamma_threshold,
+    )
+    @debug adjust_gamma_params
+
+    @info "Set structuring elements"
+    # This isn't tunable in the underlying code right now, 
+    # so just use the defaults
+    structuring_elements = IceFloeTracker.structuring_elements
```

Comment:
Not sure if this would need to be flexible as well. @danielmwatkins, what do you think?